### PR TITLE
ToricVarieties: Bug fix

### PR DESCRIPTION
**Pull request by HereAround**

### CODE REVIEW

**Comment by HereAround** [comment]
Fixes for bugs reported by @lkastner:
- rank of character lattice uses ambient dimension of fan,
- `hastorusfactor` compares the ambient dimension of the fan to the rank of (matrix formed from) the rays,
- proper restriction to image in computation to Picard group (via https://github.com/oscar-system/Oscar.jl/pull/1025).

Furthermore, use logic: For affine toric varieties, the Picard group is trivial. Use this to set this attribute upon construction.

File: src/ToricVarieties/NormalToricVarieties/properties.jl [hunk 1]
```diff
@@ -152,7 +152,7 @@ false
 """
 function hastorusfactor(v::AbstractNormalToricVariety)
     return get_attribute!(v, :hastorusfactor) do
-        return pm_object(v).FAN_DIM < pm_object(v).FAN_AMBIENT_DIM
+                return Polymake.common.rank(rays(fan(v))) < ambient_dim(fan(v))
```
Comment (fingolfin) [inline src/ToricVarieties/NormalToricVarieties/properties.jl]: Why the whitespace change? (not that it matters / is important, feel free to ignore)
```suggestion
        return Polymake.common.rank(rays(fan(v))) < ambient_dim(fan(v))
```

**Review by fingolfin — APPROVED** [review]
Looks good to me but probably @lkastner should review?

File: src/ToricVarieties/NormalToricVarieties/properties.jl (hunk 1 above)
Comment (HereAround) [reply src/ToricVarieties/NormalToricVarieties/properties.jl]: That was not intended. Let me check.

Comment (HereAround) [reply src/ToricVarieties/NormalToricVarieties/properties.jl]: Ok. This should be fixed now.

**Comment by HereAround** [comment]
> Looks good to me but probably @lkastner should review?

Yes, agreed.

**Review by lkastner — APPROVED** [review]

**Pull request merged**